src/components/Payroll/PayrollConfiguration/PayrollConfiguration.tsx

```diff
@@ -66,8 +71,13 @@ export const Root = ({ onEvent, companyId, payrollId, dictionary }: PayrollConfi
     onEvent(componentEvents.RUN_PAYROLL_BACK)
   }
   const onCalculatePayroll = async () => {
-    await mutate()
-    onEvent(componentEvents.RUN_PAYROLL_CALCULATED)
+    await calculatePayroll({
+      request: {
+        companyId,
+        payrollId,
+      },
+    })
+    await refetch()
```

Comment:
Do we still need to refetch even after the update we made to invalidate the cache?

Reply:
You're right, we shouldn't. I'll double check

Reply:
I removed it